src/inference/src/core.cpp

```diff
@@ -48,12 +45,17 @@ class Core::Impl : public CoreImpl {
 Core::Core(const std::string& xml_config_file) {
     _impl = std::make_shared<Impl>();
 
+    std::string xmlConfigFile = ov::findPluginXML(xml_config_file);
+    if (!xmlConfigFile.empty())
+        OV_CORE_CALL_STATEMENT(
+            // If XML is default, load default plugins by absolute paths
+            _impl->register_plugins_in_registry(xmlConfigFile, xml_config_file.empty());)
 #ifdef OPENVINO_STATIC_LIBRARY
+    // Load statically linked plugins
     OV_CORE_CALL_STATEMENT(_impl->register_plugins_in_registry(::getStaticPluginsRegistry());)
 #else
-    OV_CORE_CALL_STATEMENT(
-        // If XML is default, load default plugins by absolute paths
-        _impl->register_plugins_in_registry(findPluginXML(xml_config_file), xml_config_file.empty());)
+    // Load plugins from pre-compiled list
+    OV_CORE_CALL_STATEMENT(_impl->register_compile_time_plugins();)
 #endif
```

Comment:
I suppose after this alignment where both static and dynamic build use ie_plugins_hpp with prebuilt list of plugins, we don't need conditional compilation with `OPENVINO_STATIC_LIBRARY` macro

Reply:
Currently, we need this macro a few times as the data structures differ a bit (still we just removed an unnecessary use from `core_impl.cpp`). We think it could be possible to move the whole static/shared-dependent stuff to `ie_plugins.hpp`, including moving the `PluginDescriptor` struct to there. This would result in a unified `register_compile_time_plugins` function called inside the `Core` class that handles everything else: In the static library case, the plugin descriptor struct will be initialized using the create functions, otherwise the shared libraries are registered. However, we think this is rather disruptive, even more disruptive as the PR already is - but probably possible. Do you think this should be implemented this way?

Reply:
I vote for unification